mws/mws.py

```diff
@@ -109,83 +102,119 @@ def clean_params(params):
     return params_enc
 
 
-def remove_namespace(xml):
+def validate_hash(response):
     """
-    Strips the namespace from XML document contained in a string.
-    Returns the stripped string.
+    Input is a requests.response object, see the test class FakeResponse.
     """
-    regex = re.compile(' xmlns(:ns2)?="[^"]+"|(ns2:)|(xml:)')
-    return regex.sub('', xml)
+    hash_ = utils.calc_md5(response.content)
+    if response.headers['content-md5'].encode() != hash_:
+        raise MWSError("Wrong Content length, maybe amazon error...")
 
 
-class DictWrapper(object):
-    """
-    Main class that converts XML data to a parsed response object as a tree of ObjectDicts,
-    stored in the .parsed property.
```

Comment:
It was supposed to be the .parsed property before too. And the syntax: response.parsed reads nice.

Reply:
here's a quick example of what I mean:
```python
import mws

SECRET_KEY = "nCDKnLQedNfw7NyAnyB48XtY07+fm6jCoVadkj56"
AWS_ACCESS_KEY = "AKIAJCXLZCMLO6VS24TA"
SELLER_ID = "A27DY0XIZK94N5"
MARKETPLACE_ID = "A1F83G8C2ARO7P"

# Setup - the same for both
products_api = mws.Products(access_key=AWS_ACCESS_KEY, secret_key=SECRET_KEY, account_id=SELLER_ID, region='UK')
asin = 'B0107YJU8Q'

competitive_prices = products_api.get_competitive_pricing_for_asin(
    asins=[asin], marketplace_id=MARKETPLACE_ID)


product = competitive_prices.parsed

# For the current version, including the develop branch, this will return the ASIN of the product by using the dot
# notation, the .get() method and a deeper dot notation
# Access the ASIN:
try:
    print(product.ASIN)
except KeyError as exc:
    print('Cannot access this by dot notation')
print(product.get('ASIN')) # this returns the ASIN
print(product.Product.Identifiers.MarketplaceASIN.ASIN) # This returns the ASIN


# xmltodict version
# Sadly, this version does not return the ASIN as it is now @ASIN 
try:
    print(product.ASIN) # This is returning None, should retrun the ASIN
except Exception as exc:
    print('Cannot access the .ASIN')

print(product.get('ASIN')) # This returns None, should retrun the ASIN
print(product.Product.Identifiers.MarketplaceASIN.ASIN)
print(product.get('@ASIN')) # Returns the ASIN, its not very 'clean'

## There are other areas that the same @ symbol is prefixing keys, such as offer listings etc

```


We are now getting elements returned prefiex with  `@`  which breaks the API.
I feel it should keep the consistent dot notation so it works with all nodes, not just some.

so you should be able to access any node using the dot notation, and not have to change back to the `.get('@ASIN)` , `.get('@belongsToRequester)` etc:

```
{'CompetitivePrice': {'@belongsToRequester': 'false', '@condition': 'New', '@subcondition': 'New', '@xmlns': OrderedDict([('ns2', 'http://mws.amazonservices.com/schema/Products/2011-10-01/default.xsd')]), 'CompetitivePriceId': '1', 'Price': {'LandedPrice': {'CurrencyCode': 'GBP', 'Amount': '99.00'}, 'ListingPrice': {'CurrencyCode': 'GBP', 'Amount': '99.00'}, 'Shipping': {'CurrencyCode': 'GBP', 'Amount': '0.00'}}}}
```

I think that the dot notation either works fully (so you can get access to whichever bit of the returned data you want, or gets dropped completely and the dictionary method is used (so .get() and data['key'] etc)

Does that make sense?

Reply:
I've done a bit of digging and it looks like the xmltodict library won't let us achieve this:

https://github.com/martinblech/xmltodict/blob/master/xmltodict.py#L204

Which is a shame.

I still think its worth a discussion on if this is a breaking change we are happy to use.